doc/dev/testing.md

```diff
@@ -0,0 +1,35 @@
+This document explains how to run tests for etcd-operator functionality
```

Comment:
Can you rename this file to azure_testing.md ?

Reply:
This document explains how to run tests for azure backup storage feature.

Reply:
I created the doc to also mention how to run unit tests in general and supplying a link to the e2e testing docs... but if preferred, I can remove that info and rename/dedicate this to azure testing

Reply:
Ah I see.

That would be nice. But this doc only talks about Azure thing. That is confusing me.

Could you submit the whole doc (in a different PR maybe)?

Reply:
I've re-named the file and devoted it to Azure/ABS testing for now...